frontend/src/container/AdvancedSearch.jsx

```diff
@@ -175,6 +187,167 @@ export class AdvancedSearch extends React.Component {
+  }
+
+  handleChangeCreatedRange = (dateObject) => {
+    const { props, state } = this
+    let currentSearch
+
+    if (state.searchType === ADVANCED_SEARCH_TYPE.CONTENT) {
+      currentSearch = props.contentSearch
+    }
+    /*
+      if (state.searchType === ADVANCED_SEARCH_TYPE.USER) {
+        currentSearch= props.userSearch
+      }
+
+      if (state.searchType === ADVANCED_SEARCH_TYPE.SPACE) {
+        currentSearch = props.spaceSearch
+      }
+    */
+
+    let newAppliedCreatedRange = {}
+    const oldAppliedCreatedRange = currentSearch.appliedFilterList.createdRange
+
+    if (oldAppliedCreatedRange) {
+      if (Object.keys(dateObject)[0] in oldAppliedCreatedRange) {
+        Object.keys(oldAppliedCreatedRange).forEach(key => {
+          if (key !== Object.keys(dateObject)[0]) {
+            newAppliedCreatedRange[key] = oldAppliedCreatedRange[key]
+          }
+        })
+      } else newAppliedCreatedRange = { ...oldAppliedCreatedRange, ...dateObject }
+    } else newAppliedCreatedRange = dateObject
+
+    props.dispatch(setAppliedFilter(ADVANCED_SEARCH_FILTER.CREATED_RANGE, newAppliedCreatedRange, state.searchType))
+
+    this.getSearchResult(
+      { ...currentSearch, searchType: state.searchType },
+      currentSearch.resultList.length,
+      currentSearch.appliedFilterList.searchFieldList,
+      newAppliedCreatedRange,
+      currentSearch.appliedFilterList.modifiedRange,
+      currentSearch.appliedFilterList.searchFacets
+    )
+  }
+
+  handleChangeModifiedRange = (dateObject) => {
+    const { props, state } = this
+    let currentSearch
+
+    if (state.searchType === ADVANCED_SEARCH_TYPE.CONTENT) {
+      currentSearch = props.contentSearch
+    }
+    /*
+      if (state.searchType === ADVANCED_SEARCH_TYPE.USER) {
+        currentSearch= props.userSearch
+      }
+
+      if (state.searchType === ADVANCED_SEARCH_TYPE.SPACE) {
+        currentSearch = props.spaceSearch
+      }
+    */
+
+    let newAppliedModifieddRange = {}
+    const oldAppliedModifieddRange = currentSearch.appliedFilterList.modifiedRange
+
+    if (oldAppliedModifieddRange) {
+      if (Object.keys(dateObject)[0] in oldAppliedModifieddRange) {
+        Object.keys(oldAppliedModifieddRange).forEach(key => {
+          if (key !== Object.keys(dateObject)[0]) {
+            newAppliedModifieddRange[key] = oldAppliedModifieddRange[key]
+          }
+        })
+      } else newAppliedModifieddRange = { ...oldAppliedModifieddRange, ...dateObject }
+    } else newAppliedModifieddRange = dateObject
+
+    props.dispatch(setAppliedFilter(ADVANCED_SEARCH_FILTER.MODIFIED_RANGE, newAppliedModifieddRange, state.searchType))
+
+    this.getSearchResult(
+      { ...currentSearch, searchType: state.searchType },
+      currentSearch.resultList.length,
+      currentSearch.appliedFilterList.searchFieldList,
+      currentSearch.appliedFilterList.createdRange,
+      newAppliedModifieddRange,
+      currentSearch.appliedFilterList.searchFacets
+    )
```

Comment:
This is the same code as the one at https://github.com/tracim/tracim/pull/4205/files#diff-b55d31b2a6da3b235389870d0f8c87611170ff6eb6e3c4e7b81215b31543e36cR246

Maybe use a function? This will allow naming this chunk of code, avoid duplication, and spot what is different between the two thanks to the function parameters.

Reply:
fix